sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java

```diff
@@ -306,6 +307,36 @@ public DataLakeFileAsyncClient getFileAsyncClient(String fileName) {
         }
     }
 
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DataLakeFileAsyncClient> createFileIfNotExists(String fileName) {
+        return createFileIfNotExistsWithResponse(fileName, null, null, null, null).flatMap(FluxUtil::toMono);
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DataLakeFileAsyncClient>> createFileIfNotExistsWithResponse(String fileName, String permissions,
+                                                                          String umask, PathHttpHeaders headers, Map<String, String> metadata) {
+        return createFileIfNotExistsWithResponse(fileName, permissions, umask, headers, metadata, null);
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DataLakeFileAsyncClient>> createFileIfNotExistsWithResponse(String fileName, String permissions, String umask, PathHttpHeaders headers, Map<String, String> metadata, Context context) {
+        DataLakeRequestConditions requestConditions = new DataLakeRequestConditions().setIfNoneMatch(Constants.HeaderConstants.ETAG_WILDCARD);
+            DataLakeFileAsyncClient dataLakeFileAsyncClient = getFileAsyncClient(fileName);
+            return dataLakeFileAsyncClient.createWithResponse(permissions, umask, pathResourceType,
```

Comment:
I think you can return dataLakeFileAsyncClient.createIfNotExistsWithResponse instead and then you don't have to duplicate the logic of checking again.

Reply:
This should apply to delete as well